### PR TITLE
[FX-1716] Constrain collection's featured artists

### PR DESCRIPTION
**Pull request by anandaroop**
**_WIP because this needs a rebase once #2037 is merged_**

https://artsyproduct.atlassian.net/browse/FX-1716

Paired with @pepopowitz on this.

When a collection definition includes a list of artist IDs, we don't want the Featured Artists in the header to stray beyond those artists.

This PR adds logic to constrain that list of Featured Artists, similarly to how the counterpart component in Reaction already does it.

In this example, from the [**Black Arts Movement**](https://staging.artsy.net/collection/black-arts-movement) collection ([MP query](https://metaphysics-staging.artsy.net/v2?query=%7B%0A%20%20marketingCollection(slug%3A%20%22black-arts-movement%22)%20%7B%0A%20%20%20%20title%0A%20%20%20%20query%20%7B%0A%20%20%20%20%20%20artistIDs%0A%20%20%20%20%7D%0A%20%20%20%20artworksConnection(first%3A%201%2C%20aggregations%3A%20%5BMERCHANDISABLE_ARTISTS%2C%20MEDIUM%2C%20MAJOR_PERIOD%2C%20TOTAL%5D)%20%7B%0A%20%20%20%20%20%20merchandisableArtists%20%7B%0A%20%20%20%20%20%20%20%20internalID%0A%20%20%20%20%20%20%20%20name%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D)), the Featured Artists would have included the non-requested artist Camille Billops, but will now correctly filter out that entry:

banda

I've also roughed in a Storybook story for Collections in this PR, as that seemed the most convenient way to sample a few different collections during development.

### CODE REVIEW

**Comment by ashleyjelks** [comment]
Looks good to me! Merge when ready!

File: src/lib/Scenes/Collection/__stories__/Collection.story.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const CollectionStoryRenderer = ({ collectionID }) => {
+  return (
+    <CollectionRenderer
+      collectionID={collectionID}
+      render={renderWithLoadProgress(CollectionContainer, { collectionID })}
+    />
+  )
+}
+
+storiesOf("Collection")
```
Comment (anandaroop) [inline src/lib/Scenes/Collection/__stories__/Collection.story.tsx]: This simply renders a handful of different collections that are known to be used during development currently. 

This seems to be one use of Storybooks within this repo (see e.g. fairs and shows), in addition to the more typical use of developing components in isolation (see e.g. artworks).

👇 

coll

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>
<b>🃏 FAIL</b> in <a href="https://github.com/artsy/emission/blob/1716-constrain-to-explicit-artist-ids-list/src/lib/Scenes/Collection/Screens/__tests__/CollectionHeader-tests.tsx">CollectionHeader-tests.tsx</a>


<li>
<a href='https://github.com/artsy/emission/blob/1716-constrain-to-explicit-artist-ids-list/src/lib/Scenes/Collection/Screens/__tests__/CollectionHeader-tests.tsx#L31'>renders properly</a>
<br/>


<details>
<summary>Full message</summary>
<pre><code>
TypeError: Cannot read property 'replace' of undefined
    at /home/travis/build/artsy/emission/node_modules/enzyme/build/ReactWrapper.js:906:58
    at Array.map (<anonymous>)
    at ReactWrapper.<anonymous> (/home/travis/build/artsy/emission/node_modules/enzyme/build/ReactWrapper.js:905:37)
    at ReactWrapper.single (/home/travis/build/artsy/emission/node_modules/enzyme/build/ReactWrapper.js:1720:25)
    at ReactWrapper.html (/home/travis/build/artsy/emission/node_modules/enzyme/build/ReactWrapper.js:897:21)
    at html (/home/travis/build/artsy/emission/src/lib/Scenes/Collection/Screens/__tests__/CollectionHeader-tests.tsx:31:15)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:271:22)
    at Generator.prototype.<computed> [as next] (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:97:21)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:45:40)
    at invoke (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:135:20)
    at /home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:145:13
    at /home/travis/build/artsy/emission/node_modules/react-tracking/node_modules/core-js/modules/es.promise.js:113:22
    at flush (/home/travis/build/artsy/emission/node_modules/react-tracking/node_modules/core-js/internals/microtask.js:26:9)
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
</code></pre>
</details>
</li>
<br/>

</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by artsyit** [comment]
:rocket: PR was released in v1.21.9 :rocket:

**Pull request merged**